features/include/pcl/features/impl/good.hpp

```diff
@@ -0,0 +1,666 @@
+    p.z = YOZ_projected_view ->points.at(i).z + z;
+    
+    //if adaptive_support_lenght parameter == false, some points might be projected outside of the plane, we must discard them.
+    if ((trunc(p.y / interval_y) < YOZ_histogram.size()) and (trunc(p.z / interval_z) < YOZ_histogram.at(0).size())
+	  and (trunc(p.y / interval_y) >= 0) and (trunc(p.z / interval_z) >= 0))
+    {
+      YOZ_histogram.at(trunc(p.y / interval_y)).at(trunc(p.z / interval_z))++;
+    }
+  }    
+}
+//////////////////////////////////////////////////////////////////////////////////////////////
+
+template <typename PointInT, int NumberOfBins>
+void 
+pcl::GOODEstimation<PointInT, NumberOfBins>::create2DHistogramFromXOZProjection (PointCloudIn  XOZ_projected_view, double largest_side, 
+				      unsigned int number_of_bins, int sign, std::vector<std::vector<unsigned int> > &XOZ_histogram)
+{  
+  double x = largest_side / 2;
+  double z = largest_side / 2; 
+  double interval_x = largest_side / number_of_bins; 
+  double interval_z = largest_side / number_of_bins; 
+  for (size_t i = 0; i < XOZ_projected_view->points.size(); i++)
+  {
+    pcl::PointXYZ p;
+    p.x = sign *  XOZ_projected_view->points.at(i).x +  x;    
+    p.z = XOZ_projected_view->points.at(i).z + z;		
+    
+    if ((trunc(p.x / interval_x) < XOZ_histogram.size()) and (trunc(p.z / interval_z) < XOZ_histogram.at(0).size())
+	  and (trunc(p.x / interval_x) >= 0) and (trunc(p.z / interval_z) >= 0))
+    {
+      XOZ_histogram.at(trunc(p.x / interval_x)).at(trunc(p.z / interval_z))++;
+    }
+  }  
+}
+//////////////////////////////////////////////////////////////////////////////////////////////
+
+template <typename PointInT, int NumberOfBins>
+void 
+pcl::GOODEstimation<PointInT, NumberOfBins>::create2DHistogramFromXOYProjection (PointCloudIn  XOY_projected_view,	double largest_side,
+				      unsigned int number_of_bins, int sign, std::vector<std::vector<unsigned int> > &XOY_histogram)
+{    
+  double x = largest_side / 2; 
+  double y = largest_side / 2; 
+  double interval_x = largest_side / number_of_bins; 
+  double interval_y = largest_side / number_of_bins; 
+  for (size_t i = 0; i < XOY_projected_view->points.size(); i++)
+  {
+    pcl::PointXYZ p;
+    p.x = sign * XOY_projected_view->points.at(i).x + x;
+    p.y = sign * XOY_projected_view->points.at(i).y + y;        
+    if ((trunc(p.x / interval_x) < XOY_histogram.size()) and (trunc(p.y / interval_y) < XOY_histogram.at(0).size()) and 
+	  (trunc(p.x / interval_x) >= 0) and (trunc(p.y / interval_y) >= 0))
+    {
+	XOY_histogram.at(trunc(p.x / interval_x)).at(trunc(p.y / interval_y))++;
+    }
+  }
+}
+//////////////////////////////////////////////////////////////////////////////////////////////
+
+template <typename PointInT, int NumberOfBins>
+void
+pcl::GOODEstimation<PointInT, NumberOfBins>::normalizingHistogram (std::vector<unsigned int> histogram, std::vector<float> &normalized_histogram)
+{
+  int sum_all_bins = 0;    
```

Comment:
Why are you summing unsigned ints into a signed int? Besides, later you cast it into float anyway.